module_utils/common_utils.py

```diff
@@ -26,6 +26,19 @@ def check_if_valid_ip(ip_address):
     except:
         return False
 
+def traverse_and_retrieve_value(module, object , attribute_list):
+    '''
+    params:
+    - object: Object where value is to be searched from attribute list
+    - attribute_list: List to be used for searching attribute value
+    '''
+    for attribute in attribute_list:
+        if object.__contains__(attribute):
+            object = object[attribute]
+        else:
+            module.fail_json(msg='Error while  traversing and retrieving value. Attribute named %s was not found.'% attribute)
```

Comment:
raise exception and catch in parent function.

Reply:
Done.